packages/server/lib/server-base.ts

```diff
@@ -216,7 +242,19 @@ export abstract class ServerBase<TSocket extends SocketE2E | SocketCt> {
         return this._getRemoteState()
       }
 
-      this.createNetworkProxy({ config, getCurrentBrowser, getRemoteState, shouldCorrelatePreRequests })
+      const getRemoteStateFor = (url) => {
+        return this._getRemoteStateFor(url)
+      }
+
+      const getOriginStack = () => {
+        return this._originStack
+      }
+
+      const resetRemoteState = () => {
+        this._resetRemoteState()
+      }
```

Comment:
What do you think about `RemoteState` being its own class? Then instead of attaching more methods to `Server`, the remote-stated-related props and methods could be in that class and then the server and proxy middleware just have a `remoteState` property that's an instance of the class.

Reply:
Added RemoteStates class. [556a615](https://github.com/cypress-io/cypress/pull/20752/commits/556a615bcc29fac5f31435c292f72575a73e8b19)